pymc3/model.py

```diff
@@ -1049,10 +1074,14 @@ def make_obs_var(
             self.add_random_variable(observed_rv_var, dims)
             self.observed_RVs.append(observed_rv_var)
 
+            # Create deterministic that combines observed and missing
             rv_var = at.zeros(data.shape)
             rv_var = at.set_subtensor(rv_var[mask.nonzero()], missing_rv_var)
             rv_var = at.set_subtensor(rv_var[antimask_idx], observed_rv_var)
             rv_var = Deterministic(name, rv_var, self, dims)
+            # Tag it in its own list for easier filtering
+            self.partobs_deterministics.append(rv_var)
```

Comment:
`x` is now in two lists: model.deterministics and and model.partobs_deterministics. The reason being that we want to include it in the default `sample_posterior_predictive`, and (probably) remove it from `sample` to reduce the memory footprint. This makes it easier to filter them in / out. Do you have a better suggestion as how to target these?